modules/check_in/app/services/check_in/vaos/configuration.rb

```diff
@@ -21,7 +21,15 @@ def rsa_key
 
       def connection
         Faraday.new(base_path, headers: base_request_headers, request: request_options) do |conn|
-          conn.use(:breakers, service_name:)
+                    conn.use(:breakers,
+                   service_name:,
+                   record_exceptions: [
```

Comment:
What does `record_exceptions` do? I don't think I've seen that before.

Reply:
> What does `record_exceptions` do? I don't think I've seen that before.

So Breakers keeps track of failed requests, and once the threshold is met, the "circuit breaker" is thrown. If a service is throwing a custom error that Breakers does not know about, it will ignore the error and will not throw the breaker when we might expect it (later than it should, or never if the service only throws the custom error). By adding the custom error to the exceptions list (record_exceptions), breakers will count those errors towards the threshold, and throw the breaker appropriately. My thought was to try this with one service to see if it works, before applying it to all services, and since VAOS is know to report inaccurately, I thought it would be a good one.

Reply:
I don't see `record_exceptions` in Breakers or Faraday so I'm not sure it's going to know what to do with this.
> If a service is throwing a custom error that Breakers does not know about, it will ignore the error and will not throw the breaker when we might expect it

I didn't realize this 🤔 which errors _does_ it know about?

But I definitely think it makes sense to try on VAOS, since we know that service is reliably chaotic when it come to this bug.